### PR TITLE
Feature NXP-25831 big file optimization

### PR DESCRIPTION
**Pull request by kevinleturc**
Waiting QA to be stable enough to run a T&P.

### CODE REVIEW

File: nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/NginxConstants.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public static final String X_ACCEL_LOCATION_HEADER = "X-Accel-Location";
+
+    public static final String X_ACCEL_REDIRECT_HEADER = "X-Accel-Redirect";
+
+    public static final String X_REQUEST_BODY_FILE_HEADER = "X-Request-Body-File";
+
+    public static final String X_CONTENT_MD5_HEADER = "X-Content-MD5";
+
+    private NginxConstants() {
+        // instantiation allowed
```
Comment (efge) [inline nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/NginxConstants.java]: disallowed.

File: nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadServiceImpl.java [hunk 2]
```diff
@@ -545,6 +549,29 @@ public void downloadBlob(HttpServletRequest request, HttpServletResponse respons
...
+                                                   .relativize(blob.getFile().toURI())
+                                                   .getPath();
+
+                    if (xAccelLocation.endsWith("/")) {
+                        xAccelLocation = xAccelLocation + relative;
+                    } else {
+                        xAccelLocation = xAccelLocation + "/" + relative;
+                    }
+
+                    response.setHeader(NginxConstants.X_ACCEL_REDIRECT_HEADER, xAccelLocation);
```
Comment (efge) [inline nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadServiceImpl.java]: Can you remove the 2 empty lines above? The code is already sparse enough

File: nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/NginxConstants.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+package org.nuxeo.ecm.core.io;
+
+/**
+ * @since 10.3
+ */
+public final class NginxConstants {
+
+    /** nuxeo.conf parameter to enable X-Accel in Nuxeo Platform. */
+    public static final String X_ACCEL_ENABLED = "nginx.x.accel.enabled";
```
Comment (efge) [inline nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/NginxConstants.java]: Could you rename to `nuxeo.nginx.accel.enabled `?

**Comment by efge** [comment]
👍

**Comment by kevinleturc** [comment]
On its way: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-kleturc-2/132/

**Comment by troger** [comment]
:+1:

**Pull request merged**